bundles/org.openhab.binding.dolbycp/src/main/java/org/openhab/binding/dolbycp/internal/DolbyCPHandler.java

```diff
@@ -0,0 +1,314 @@
+                            client.setFader(((DecimalType) command).intValue());
+                        }
+                        if (command instanceof IncreaseDecreaseType) {
+                            client.setFaderDelta(command == IncreaseDecreaseType.INCREASE ? 1 : -1);
+                        }
+                    }
+                }
+            } catch (IOException e) {
+                disposeClient(e);
+            }
+        }
+    }
+
+    @Override
+    public void initialize() {
+        final DolbyCPConfiguration config = getConfigAs(DolbyCPConfiguration.class);
+        this.config = config;
+        updateStatus(ThingStatus.UNKNOWN);
+
+        scheduler.execute(() -> {
+            try {
+                ScheduledFuture<?> scheduleFuture = this.scheduleFuture;
+                this.scheduleFuture = null;
+                if (scheduleFuture != null) {
+                    scheduleFuture.cancel(true);
+                }
+
+                CP750Client client = this.client;
+                this.client = null;
+                if (client != null) {
+                    client.close();
+                }
+
+                client = new CP750Client(config.hostname, config.port);
+                this.client = client;
+                for (CP750Field field : CP750Field.values()) {
+                    client.addListener(field, this);
+                }
+                client.refresh();
+                updateStatus(ThingStatus.ONLINE);
+
+                // Start scheduler
+                if (config.refreshInterval > 0) {
+                    this.scheduleFuture = scheduler.scheduleWithFixedDelay(this::refresh, config.refreshInterval,
+                            config.refreshInterval, TimeUnit.SECONDS);
+                }
+            } catch (IOException e) {
+                disposeClient(e);
+            }
+        });
+    }
+
+    private void disposeClient(@Nullable Exception e) {
+        ScheduledFuture<?> scheduleFuture = this.scheduleFuture;
+        this.scheduleFuture = null;
+        if (scheduleFuture != null) {
+            try {
+                scheduleFuture.cancel(true);
+            } catch (Exception ignore) {
+            }
+        }
+        CP750Client client = this.client;
+        this.client = null;
+        if (client != null) {
+            client.removeListener(this);
+            try {
+                client.close();
+            } catch (Exception ignore) {
+            }
+        }
+        if (e == null) {
+            updateStatus(ThingStatus.OFFLINE);
+        } else {
+            DolbyCPConfiguration config = this.config;
+            String hostname = config == null ? "unknown" : config.hostname;
+            int port = config == null ? -1 : config.port;
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                    "Communication error to " + hostname + ":" + port);
+            logger.error("CP750 on {}:{}", hostname, port, e);
```

Comment:
Checking config and setting the Thing status accordinlgy is usually done at `initialize`.
Also the error log should not be deed as the Thing StatusDetail is allready set. It would also be usefull to distinct between valid config (CONFIG_ERROR) and non-reachable host/ (COMMUNICATION_ERROR)

Reply:
Rewrote the whole part